### ISSUE TITLE
Not remove 'del(.metadata.annotations.kubectl.kubernetes.io/last-applied-configuration)'

### ISSUE THREAD

**Issue by jniltinho** [mikefarah/yq#772]
**Describe the bug**

Note that any how to questions should be posted in the discussion board and not raised as an issue.

Version of yq: 4.6.3
Operating system: linux
Installed via: binary

**Input Yaml**
Concise yaml document(s) (as simple as possible to show the bug, please keep it to 10 lines or less)

data1.yml:
```apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
    kubectl.kubernetes.io/last-applied-configuration: |
      {"casssaaaa"}
  name: postgres-db
  namespace: postgres-dev
spec:
  progressDeadlineSeconds: 600
  replicas: 0
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: postgres-db
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 0
    type: RollingUpdate
  template:
    metadata:
      labels:
        app: postgres-db
    spec:
      affinity: {}
      containers:
        - env:
            - name: POSTGRES_PASSWORD
              value: casa
            - name: POSTGRES_PORT
              value: "5432"
            - name: POSTGRES_USER
              value: user1
          image: postgres
          imagePullPolicy: Always
          name: postgres-db
          ports:
            - containerPort: 5432
              name: 5432tcp02
              protocol: TCP
          resources: {}
          securityContext:
            allowPrivilegeEscalation: false
            privileged: false
            readOnlyRootFilesystem: false
            runAsNonRoot: false
          stdin: true
          terminationMessagePath: /dev/termination-log
          terminationMessagePolicy: File
          tty: true
      dnsPolicy: ClusterFirst
      hostname: db
      imagePullSecrets:
        - name: docker
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
```

**Command**
The command you ran:
```
yq eval 'del(.metadata.annotations.kubectl.kubernetes.io/last-applied-configuration)'  data1.yml
```

**Comment by mikefarah** [comment]
the keys have dots (.) in them, you will need to use to use the surround the key name with [] as described here: https://mikefarah.gitbook.io/yq/operators/traverse-read#special-characters

**Comment by sjbylo** [comment]
Hi, 

pls can you help me get the third example, below to work?

```
$ yq --version
yq (https://github.com/mikefarah/yq/) version 4.14.1
$ echo '"{}": frog' | yq eval '.["{}"]' -
frog
$ echo '"a.b": frog' | yq eval '.["a.b"]' -
frog
$ echo 'a.b: frog' | yq eval '.[a.b]' -
Error: Parsing expression: Lexer error: could not match text starting at 1:3 failing at 1:5.
        unmatched text: "a."

```

I am trying to eval this:

```
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
```

**Comment by Markieta** [comment]
> the keys have dots (.) in them, you will need to use to surround the key name with [] as described here: https://mikefarah.gitbook.io/yq/operators/traverse-read#special-characters

**TL;DR**:

```bash
yq eval 'del(.metadata.annotations.["kubectl.kubernetes.io/last-applied-configuration"])' data1.yml
```
